modules/database/database.bal

```diff
@@ -399,7 +399,7 @@ public isolated transactional function getPluginEndpoint(int endpointId) returns
 
 public isolated transactional function addPluginEndpoint(*PluginEndpoint endpoint) returns PluginEndpointFull|error {
     var result = check experimentDB->execute(
-        `INSERT INTO PluginEndpoints (url, type) VALUES (${endpoint.url}, ${endpoint.'type});`
+        `INSERT INTO PluginEndpoints (url, type) VALUES (${endpoint.url}, ${endpoint.'type}) WHERE NOT EXISTS(SELECT * FROM PluginEndpoints WHERE url=${endpoint.url} and type=${endpoint.'type});`
```

Comment:
It was actually intentional to let the database ensure the uniqueness of the url column. If there is a unique constraint missing it should be added in the liquibase config. This is a workaround not a fix.